node/chainSimulator/configs/configs.go

```diff
@@ -130,9 +111,80 @@ func CreateChainSimulatorConfigs(args ArgsChainSimulatorConfigs) (*ArgsConfigsSi
 		Configs:               configs,
 		ValidatorsPrivateKeys: privateKeys,
 		GasScheduleFilename:   gasScheduleName,
+		InitialWallets:        initialWallets,
 	}, nil
 }
 
+func generateGenesisFile(args ArgsChainSimulatorConfigs, configs *config.Configs) (*dtos.InitialWalletKeys, error) {
+	addressConverter, err := factory.NewPubkeyConverter(configs.GeneralConfig.AddressPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	initialWalletKeys := &dtos.InitialWalletKeys{
+		ShardWallets: make(map[uint32]*dtos.WalletKey),
+	}
+
+	initialAddressWithStake, err := generateWalletKeyForShard(0, args.NumOfShards, addressConverter)
```

Comment:
might extract the 0 into a const. Something like: walletShardIdForStake

Reply:
added